### PR TITLE
=str #17854: fullClose should not Abort

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by jrudolph** [comment]
How about splitting it up into two commits, one that fixes the `Abort` problem and one that fixes the `ConfirmedClose` / `Close` sequence problem in `fullClose` mode.

File: akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala [hunk 1]
```diff
@@ -154,17 +142,25 @@ private[akka] abstract class TcpStreamActor(val settings: ActorMaterializerSetti
...
-          connection ! Close
-          tryShutdown()
-        } else
-          connection ! ConfirmedClose
+
+        if (halfClose) {
+          if (tcpInputs.isClosed) {
+            // Reading has stopped, either because of cancel, or PeerClosed, just Close now
+            connection ! Close
+            tryShutdown()
```
Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala]: These two lines now show up several times. Maybe it should be given a name now... :)

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala]: well, three times :) I add a name.

**Comment by drewhk** [comment]
I am not sure it worth an extra commit, but why not.

**Comment by ktoso** [comment]
LGTM, validated on my machine a few hundred times.

**Comment by drewhk** [comment]
extra Jenkins run won't hurt

**Pull request merged**